### PR TITLE
Add option --hex-dump-one-line to print the hex dump output in one line

### PR DESCRIPTION
**Pull request by rastogishubham**
Based on Adrian's comments on https://github.com/apple/llvm-project/pull/5140 I needed to add an option to print the hex dump output in one line. This PR addresses that

### CODE REVIEW

**Comment by rastogishubham** [comment]
@swift-ci please test

File: llvm/tools/llvm-cas-dump/MCCASPrinter.cpp [hunk 1]
```diff
@@ -79,13 +79,21 @@ Error MCCASPrinter::printMCObject(MCObjectProxy MCObj, DWARFContext *DWARFCtx) {
...
+          data.take_front(data.size()), OS,
           [this](unsigned char c) { OS << llvm::format_hex(c, 4); }, " ");
       OS << "\n";
-      data = data.drop_front(data.size() < 8 ? data.size() : 8);
+    } else {
+      while (!data.empty()) {
+        OS.indent(Indent);
+        llvm::interleave(
+            data.take_front(8), OS,
+            [this](unsigned char c) { OS << llvm::format_hex(c, 4); }, " ");
```
Comment (adrian-prantl) [inline llvm/tools/llvm-cas-dump/MCCASPrinter.cpp:93]: why do we capture this here?

**Review by adrian-prantl — APPROVED** [review]

File: llvm/tools/llvm-cas-dump/MCCASPrinter.cpp (hunk 1 above)
Comment (rastogishubham) [reply llvm/tools/llvm-cas-dump/MCCASPrinter.cpp:93]: would changing it to & be better? OS is part of `MCCASPrinter` so I captured this

Comment (adrian-prantl) [reply llvm/tools/llvm-cas-dump/MCCASPrinter.cpp:93]: Ah.. it's because of OS.

Comment (adrian-prantl) [reply llvm/tools/llvm-cas-dump/MCCASPrinter.cpp:93]: we could capture `&OS` :-)
seriously, this is fine — I just overlooked OS.

Comment (rastogishubham) [reply llvm/tools/llvm-cas-dump/MCCASPrinter.cpp:93]: I will just capture OS, no problem :)

**Comment by rastogishubham** [comment]
@swift-ci please test

**Review by felipepiovezan — APPROVED** [review]

**Pull request merged**